pymc/logprob/binary.py

```diff
@@ -0,0 +1,109 @@
+#   Copyright 2023 The PyMC Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+from typing import List, Optional
+
+import numpy as np
+import pytensor.tensor as pt
+
+from pytensor.graph.basic import Node
+from pytensor.graph.fg import FunctionGraph
+from pytensor.graph.rewriting.basic import node_rewriter
+from pytensor.scalar.basic import GT, LT
+from pytensor.tensor.math import gt, lt
+
+from pymc.logprob.abstract import (
+    MeasurableElemwise,
+    MeasurableVariable,
+    _logcdf,
+    _logprob,
+)
+from pymc.logprob.rewriting import measurable_ir_rewrites_db
+from pymc.logprob.utils import ignore_logprob
+
+
+class MeasurableComparison(MeasurableElemwise):
+    """A placeholder used to specify a log-likelihood for a binary comparison RV sub-graph."""
+
+    valid_scalar_types = (GT, LT)
+
+
+@node_rewriter(tracks=[gt, lt])
+def find_measurable_comparisons(
+    fgraph: FunctionGraph, node: Node
+) -> Optional[List[MeasurableComparison]]:
+    rv_map_feature = getattr(fgraph, "preserve_rv_mappings", None)
+    if rv_map_feature is None:
+        return None  # pragma: no cover
+
+    if isinstance(node.op, MeasurableComparison):
+        return None  # pragma: no cover
+
+    (compared_var,) = node.outputs
+    base_var, const = node.inputs
+
+    if not (
+        base_var.owner
+        and isinstance(base_var.owner.op, MeasurableVariable)
+        and base_var not in rv_map_feature.rv_values
+    ):
+        return None
+
+    # Make base_var unmeasurable
```

Comment:
We have to check 2 more conditions:

1. Both sides of the comparison cannot be measurable (or even potentially) measurable, otherwise we logp would require a convolution over all possible values of both variables (which we could implement later on)
  - You can check the logic we use in Measurable Additions and Multiplications where the same logic applies (in logprob/transforms.py). In there we have some code that checks only one of the inputs is measurable. We should refactor that and reuse it here.
2. We need to check which side of the inequality is actually being measured (x < 0 vs 0 < x). Again note that the MeasurableTransform class we use has this info stored as "measurable_input_idx” or something like that

Reply:
>  In there we have some code that checks only one of the inputs is unvalued and measurable. We should refactor that into a separate function and reuse it here.

I have refactored it into a separate function and in order to test it, I just wanted to confirm the approach to make `const` measurable. Should it work if we define `const` like this?  

```
import pymc as pm
import pytensor.tensor as pt
x = pt.random.normal(2)
z = pt.random.normal(x)
y = pt.gt(x, z)

print(pm.logp(y, 1).eval())
```

Reply:
`pm.logp` should fail, because you're requesting the logp of two variables with a single value, which would require a convolution.

`pm.factorized_joint_logp` where you give a value variable to `y` and a separate one to either `x` or `z` should work.

Reply:
Can you add that as a test?